libs/asset-viewer/src/lib/state/asset-search/asset-search.effects.ts

```diff
@@ -27,208 +26,160 @@ import {
+      map(([{ query }, currentQuery]) => {
+        return actions.search({ query: { ...currentQuery, ...query } });
+      })
     )
   );
 
-  public assetStats$ = createEffect(() =>
+  public loadSelectedAsset$ = createEffect(() =>
     this.actions$.pipe(
-      ofType(actions.loadSearch),
-      switchMap(({ query }) =>
-        this.assetSearchService
-          .updateSearchResultStats(query)
-          .pipe(map((searchStats) => actions.updateStats({ searchStats })))
-      )
+      ofType(actions.selectAsset),
+      switchMap(({ assetId }) => this.assetSearchService.fetchAssetEditDetail(assetId)),
+      map((asset) => actions.setSelectedAsset({ asset }))
     )
   );
 
-  public toggleAssetDetail$ = createEffect(() => {
-    return this.actions$.pipe(
-      ofType(actions.assetClicked),
-      withLatestFrom(this.store.select(selectCurrentAssetDetail)),
-      switchMap(([{ assetId }, currentAssetDetail]) =>
-        assetId !== currentAssetDetail?.assetId
-          ? this.assetSearchService
-              .loadAssetDetailData(assetId)
-              .pipe(map((assetDetail) => actions.updateAssetDetail({ assetDetail })))
-          : of(actions.resetAssetDetail())
-      )
-    );
-  });
-
-  public showAssetDetail$ = createEffect(() => {
-    return this.actions$.pipe(
-      ofType(actions.showAssetDetail),
-      withLatestFrom(this.store.select(selectCurrentAssetDetail)),
-      switchMap(([{ assetId }, currentAssetDetail]) =>
-        assetId !== currentAssetDetail?.assetId
-          ? this.assetSearchService
-              .loadAssetDetailData(assetId)
-              .pipe(map((assetDetail) => actions.updateAssetDetail({ assetDetail })))
-          : EMPTY
-      )
-    );
-  });
-
-  public closeDetailOnUpdateSearch$ = createEffect(() => {
-    return this.actions$.pipe(
-      ofType(actions.search, actions.resetSearch, actions.removePolygon),
-      map(() => actions.resetAssetDetail())
-    );
-  });
-
-  public openSearchResults$ = createEffect(() => {
-    return this.actions$.pipe(
-      ofType(actions.updateSearchResults),
-      takeUntil(this.actions$.pipe(ofType(actions.manualToggleResult))),
-      withLatestFrom(this.store.select(selectAssetSearchNoActiveFilters)),
-      map(([_, showStudies]) => (showStudies ? actions.closeResults() : actions.openResults()))
-    );
-  });
-
-  public closeSearchResults$ = createEffect(() => {
-    return this.actions$.pipe(
-      ofType(actions.updateSearchResults),
-      withLatestFrom(this.store.select(selectAssetSearchNoActiveFilters)),
-      filter(([_, showStudies]) => showStudies),
-      map(() => actions.closeResults())
-    );
-  });
-
   public loadStudies$ = createEffect(() => {
     return this.actions$.pipe(
-      ofType(actions.initializeSearch),
-      switchMap(() => this.store.select(selectStudies)),
-      filter((x) => !x),
+      ofType(actions.initialize),
+      switchMap(() => this.store.select(selectStudies).pipe(take(1))),
```

Comment:
take(1) is not needed, right? The effect only triggers on actions.initialize, and this is only triggered once